### PR TITLE
soc/intel_adsp: support 2 and 4 DSP core cavs25 variants 

### PR DESCRIPTION
**Pull request by kv2019i**
New attempt to unblock our validation on 4 DSP core systems. Previous attempt at https://github.com/zephyrproject-rtos/zephyr/pull/40960 .

Replacing https://github.com/zephyrproject-rtos/zephyr/pull/41764 as github stopped picking up new commits from my branch to the PR.

### CODE REVIEW

**Review by andyross — DISMISSED** [review]
Still +1 from me on this one.  Still same comment: it's true that this requires a distressing amount of cut/paste in the configuration layer.  That's a failing of our existing board architecture and not something that should block merge.  We should clean things up (e.g. moving shared defaults up a level to intel_adsp/Kconfig, adding a few dts include files with variants, etc...), but as an enhancement.

**Comment by kv2019i** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/41764 is permanently stuck, replacing with this PR.

**Review by dcpleung — DISMISSED** [review]

File: boards/xtensa/intel_adsp_cavs25_tgph/intel_adsp_cavs25_tgph.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+identifier: intel_adsp_cavs25_tgph
+name: CAVS 2.5 Audio DSP for Tiger Lake H PCH (Connected Audio Voice and Speech)
+type: mcu
+arch: xtensa
+toolchain:
+  - zephyr
+testing:
+  only_tags:
+     - kernel
+     - sof
```
Comment (MaureenHelm) [inline boards/xtensa/intel_adsp_cavs25_tgph/intel_adsp_cavs25_tgph.yaml:10]: I did some cleanup here recently on the other cAVS boards, see #42748 and #42563. Please update to match

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by nashif — DISMISSED** [review]

File: boards/xtensa/intel_adsp_cavs25_tgph/intel_adsp_cavs25_tgph.yaml (hunk 1 above)
Comment (kv2019i) [reply boards/xtensa/intel_adsp_cavs25_tgph/intel_adsp_cavs25_tgph.yaml]: Oops, yes, updated!

File: dts/xtensa/intel/intel_cavs25_tgph.dtsi [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		};
+
+		idc: idc@1200 {
+			compatible = "intel,cavs-idc";
+			label = "CAVS_IDC";
+			reg = <0x1200 0x80>;
+			interrupts = <8 0 0>;
+			interrupt-parent = <&cavs0>;
+		};
+	};
```
Comment (teburd) [inline dts/xtensa/intel/intel_cavs25_tgph.dtsi:111]: This should have a sections to cover the gpdma which you've said the board supports above for testing. Without it the test case will have the warning which causes CI to fail here. Take a look at the dts/xtensa/intel/intel_cavs25.dtsi which has the ip blocks described, its likely as easy as copying and pasting it.

**Review by teburd — CHANGES_REQUESTED** [review]
Looks good to me, needs either the dma support bit removed from the yaml file or the device tree section added so that the driver can build without warnings. CI caught this.

File: dts/xtensa/intel/intel_cavs25_tgph.dtsi (hunk 2 above)
Comment (kv2019i) [reply dts/xtensa/intel/intel_cavs25_tgph.dtsi:111]: Thanks @teburd  . It seems I keep being one step behind what's happening with the main cavs25 variant. Will retry today.

Comment (kv2019i) [reply dts/xtensa/intel/intel_cavs25_tgph.dtsi:111]: Still failing, investigating: 
devicetree error: /aliases: undefined node label 'lpgpdma0'

**Review by nashif — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**